src/main.h

```diff
@@ -1627,12 +1625,15 @@ class CDiskBlockIndex : public CBlockIndex
 		{
 			READWRITE(nIsSuperBlock);
 			READWRITE(nIsContract);
-			READWRITE(sGRCAddress);
 
-                        // Blocks used to come with a reserved string. Keep (de)serializing
-                        // it until it's used.
-                        std::string sReserved;
-                        READWRITE(sReserved);
+         std::string dummy;
```

Comment:
Better way of doing this? :( Fiddled with this before. We could serialize based on block version but that removes the backward compatibility (as in when downgrading).

Reply:
It is very difficult to work with the `IMPLEMENT_SERIALIZE` and `READWRITE` macros. The underlying template code is nice.
Maybe these could be modified or broken down into separate read and write and then go with conditions.

Reply:
Yeah, I was thinking something like:
```
if(nVersion < 9) {
   // Read/write obsolete GRC address and reserved string
   std::string dummy;
   READWRITE(dummy); // Address
   READWRITE(dummy); // Reserved string
}

This will fail when downgrading if you have a chain written by a new client.